src/caches/beliefpropagationcache.jl

```diff
@@ -0,0 +1,221 @@
+default_mt_constructor(inds_e) = ITensor[denseblocks(delta(inds_e))]
+default_mt_storage(ptn::PartitionedGraph) = Dict()
```

Comment:
Again, best to avoid `mt`.

Maybe `default_cache_storage`?

Also, I would prefer to use `Dictionary()` here instead of `Dict()`, since generally I like that type better (it is faster, has a nice interface for functionality like `map`, etc.).

Reply:
Or maybe `default_message_tensors`? Since I see that inside `BeliefPropagationCache` the field storing the message tensors is called `message_tensors` which I think is a good name.